test/extension/handlers/topologymutation/handler.go

```diff
@@ -19,70 +19,239 @@ package topologymutation
+				log.Error(err, "error patching KubeadmControlPlaneTemplate")
+				return errors.Wrapf(err, "error patching KubeadmControlPlaneTemplate")
 			}
-		}
 
+		case *bootstrapv1.KubeadmConfigTemplate:
+			if err := patchKubeadmConfigTemplate(ctx, typedObj, variables); err != nil {
+				log.Error(err, "error patching KubeadmConfigTemplate")
+				return errors.Wrap(err, "error patching KubeadmConfigTemplate")
+			}
+		case *infrav1.DockerMachineTemplate:
+			if err := patchDockerMachineTemplate(ctx, typedObj, variables); err != nil {
+				log.Error(err, "error patching DockerMachineTemplate")
+				return errors.Wrap(err, "error patching DockerMachineTemplate")
+			}
+		default:
+			return nil
+		}
 		return nil
 	})
 }
 
-// patchDockerClusterTemplate patches the DockerClusterTepmlate.
-func patchDockerClusterTemplate(dockerClusterTemplate *infrav1.DockerClusterTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
-	// Get the variable value as JSON string.
-	value, err := patchvariables.GetVariableValue(templateVariables, "lbImageRepository")
+func patchKubeadmControlPlaneTemplate(ctx context.Context, kubeadmControlPlaneTemplate *controlplanev1.KubeadmControlPlaneTemplate, templateVariables map[string]apiextensionsv1.JSON) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	// 1) If the Kubernetes version from builtin.controlPlane.version is below 1.24.0 set "cgroup-driver": "cgroupfs" to
+	//    - kubeadmConfigSpec.InitConfiguration.KubeletExtraArgs
+	//    - kubeadmConfigSpec.JoinConfiguration.KubeletExtraArgs
+	_, cpVersion, err := getStringValueForVariable(templateVariables, "builtin.controlPlane.version")
+	if err != nil {
+		return errors.Wrap(err, "could not set cgroup-driver to control plane template kubeletExtraArgs")
+	}
+	controlPlaneVersion, err := version.ParseMajorMinorPatchTolerant(cpVersion)
 	if err != nil {
 		return err
 	}
+	if version.Compare(controlPlaneVersion, versionCeilingForPatch) == -1 {
+		log.Info(fmt.Sprintf("Setting kubeadmControlPlaneTemplate cgroup-driver %s", "cgroup-driver"))
+		if kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.InitConfiguration != nil {
+			kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.InitConfiguration.NodeRegistration.KubeletExtraArgs["cgroup-driver"] = cgroupDriverCgroupfs
+		}
+		if kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.JoinConfiguration != nil {
+			kubeadmControlPlaneTemplate.Spec.Template.Spec.KubeadmConfigSpec.JoinConfiguration.NodeRegistration.KubeletExtraArgs["cgroup-driver"] = cgroupDriverCgroupfs
+		}
+	}
 
-	// Unquote the JSON string.
-	stringValue, err := strconv.Unquote(string(value.Raw))
+	// 2) Patch RolloutStrategy RollingUpdate MaxSurge with the value from the Cluster Topology variables.
+	//    If this is unset continue as this variable is not required.
+	if found, kcpControlPlaneMaxSurge, err := getStringValueForVariable(templateVariables, "kubeadmControlPlaneMaxSurge"); found {
```

Comment:
I think we have to check for error even if found is false.

Can we simplify this to:
```go
	found, kcpControlPlaneMaxSurge, err := getStringValueForVariable(templateVariables, "kubeadmControlPlaneMaxSurge")
	if err != nil {
		return errors.Wrap(err, "could not set KubeadmControlPlaneTemplate MaxSurge")
	}
	if found {
```